### PR TITLE
Extension/uicollectionview

### PR DESCRIPTION
**Pull request by bstien**
# Why?
Since we already have an extension for `UITableView`, it makes sense to have one for `UICollectionView` as well.

# What?
- Added an extension for `UICollectionView` so it can be reloaded with changes from the diff.
- Updated `README`.

### CODE REVIEW

File: Diffuse/Extensions/UICollectionViewExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+extension UICollectionView {
+    public func reload(with changes: CollectionChanges,
+                       section: Int = 0,
+                       updateDataSource: () -> Void) {
+        guard changes.count != 0 else { return }
+        let indexPaths = IndexPathResult(changes: changes, section: section)
+
+        performBatchUpdates({
+            updateDataSource()
```
Comment (shredlocker) [inline Diffuse/Extensions/UICollectionViewExtensions.swift:15]: Does updateDataSource have to be inside batchUpdates? I don't see why there is a closure here.

Comment (bstien) [reply Diffuse/Extensions/UICollectionViewExtensions.swift:15]: I'm not exactly sure, but I think the smartest place to update the dataSource/models is within `performBatchUpdates`. I could be wrong, seeing as I could not find any docs on this.

However, I think the main reason is thread safety. `performBatchUpdates` will run on the main thread, while the updated models could be sent to the VC on a background thread. Having `updateDataSource` within this method call will ensure everything is done on the same thread and synced.

Comment (shredlocker) [reply Diffuse/Extensions/UICollectionViewExtensions.swift:15]: 👍

File: README.md [hunk 2]
```diff
@@ -63,10 +63,10 @@ let new = [a, c, b]
 let changes = Diffuse.diff(old: old, new: new, comparator: { $0.id == $1.id })
 ```
 
-### Updating your tableView
-This extension lets you reload your `UITableView` with the changes given by the outcome of the diff. The parameter `updateDataSource` lets you update your tableView's datasource.
+### Updating your table- or collectionView
+These extensions lets you reload your `UITableView` or `UICollectionView` with the changes given by the outcome of the diff. The parameter `updateDataSource` lets you update your datasource before the changes are applied to the view.
```
Comment (vadymmarkov) [inline README.md]:
```suggestion
These extensions let you reload your `UITableView` or `UICollectionView` with the changes given by the outcome of the diff. The parameter `updateDataSource` lets you update your datasource before the changes are applied to the view.
```

**Review by vadymmarkov — APPROVED** [review]

**Pull request merged**